operation/fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/util/BulkDataImportUtil.java

```diff
@@ -31,66 +31,79 @@
-        return new FHIROperationException(msg).withIssue(ooi);
-    }
+    public BulkDataImportUtil(Parameters parameters) throws FHIROperationException {
+        if (parameters == null) {
+            throw buildExceptionWithIssue("$import parameters are empty or null", IssueType.INVALID);
+        }
 
-    public static FHIROperationException buildExceptionWithIssue(String msg, Throwable cause, IssueType issueType)
-            throws FHIROperationException {
-        OperationOutcome.Issue ooi = FHIRUtil.buildOperationOutcomeIssue(msg, issueType);
-        return new FHIROperationException(msg, cause).withIssue(ooi);
+        evaluationContext = new EvaluationContext(parameters);
     }
 
-    public static String retrieveInputFormat(Parameters parameters) throws FHIROperationException {
-        // Parameter: inputFormat (required)
-        // If there are multiple entries, the processing only takes the first entry.
-        if (parameters != null) {
-            for (Parameters.Parameter parameter : parameters.getParameter()) {
-                if (BulkDataConstants.PARAM_INPUT_FORMAT.equals(parameter.getName().getValue())
-                        && parameter.getValue() != null
-                        && parameter.getValue().is(com.ibm.fhir.model.type.String.class)) {
-                    // If the parameter isn't passed, use application/fhir+ndjson
-                    // Check the MediaType
-                    String val = parameter.getValue().as(com.ibm.fhir.model.type.String.class).getValue();
-                    if (BulkDataConstants.INPUT_FORMATS.contains(val)) {
-                        return val;
-                    }
+    /**
+     * processes the parameter inputFormat from the Parameters object and evaluationContext.
+     *
+     * @implNote If there are multiple entries, the processing only takes the first entry that matches.
+     *
+     * @return
+     * @throws FHIROperationException
+     */
+    public String retrieveInputFormat() throws FHIROperationException {
+        try {
+            Collection<FHIRPathNode> result = evaluator.evaluate(evaluationContext, "parameter.where(name = 'inputFormat').value");
+            Iterator<FHIRPathNode> iter = result.iterator();
+            while (iter.hasNext()) {
+                FHIRPathElementNode node = (FHIRPathElementNode) iter.next();
+                String val = node.asElementNode().element().as(com.ibm.fhir.model.type.String.class).getValue();
+                if (BulkDataConstants.INPUT_FORMATS.contains(val)) {
+                    return val;
                 }
             }
+        } catch (ClassCastException ce) {
+            throw buildExceptionWithIssue("$import invalid parameter name in 'inputFormat'", ce, IssueType.INVALID);
+        } catch (FHIRPathException e) {
+            throw buildExceptionWithIssue("$import invalid parameters value type in 'inputFormat'", e, IssueType.INVALID);
```

Comment:
the error messages are a little confusing to me; would something like this be better?
```suggestion
        } catch (ClassCastException ce) {
            throw buildExceptionWithIssue("$import invalid parameter value in 'inputFormat'", ce, IssueType.INVALID);
        } catch (FHIRPathException e) {
            throw buildExceptionWithIssue("Unexpected error while processing $import parameter 'inputFormat'", e, IssueType.INVALID);
```

Reply:
because I think we'd typically hit the first one when the inputFormat parameter value isn't a FHIR string and I wouldn't expect to ever hit the second one.  Or is there some case that throws FHIRPathException that I'm missing?

Reply:
That is true.